### ISSUE TITLE
Captcha image didn't refresh 

### ISSUE THREAD

**Issue by 218labs** [donovan-duplessis/cakephp-captcha#7]
Hello,
The link didn't refresh the image and when I call img/captcha.jpg it gives me always the same image by against it's ok when I call the action of the controller. 

Thank you.

**Comment by donovan-duplessis** [comment]
Hi,

Calling the controller action and it showing different images is a good start.

It is more than likely the jquery hook not being called when you click the refresh button.  Are you using the example code?

The important pieces of code is:

Controller:

`public $helpers = array(
        'Js' => array('Jquery')
    );`

View:

```
$this->Js->get('.reload')->event('click',
        "$(this).prev().attr('src', $(this).prev().attr('src') + '?' + new Date().getTime())"
    );
```

You will see above, with the jQuery click callback method, if you have class='reload' on your refresh link, that hook will be called to refresh the image.

Basically, in a nutshell, it finds the previous element (before this element is the actual image element), then reloads the image src attribute which shows a new image.

Could you possibly show me parts of your view so I can help you fix this issue?

Regards,
Donovan

**Comment by 218labs** [comment]
Hi Donovan,

This is my web page : http://www.netgroup-apps.com/ralitrack/eng/users/add the captcha is at the buttom.

Link to action captcha in users controller :
http://www.netgroup-apps.com/ralitrack/users/captcha?url=captcha

Links to the view and the controller :

http://www.netgroup-apps.com/ralitrack/UsersController.txt
http://www.netgroup-apps.com/ralitrack/add.txt

Regards,
Abderrahmane

**Comment by donovan-duplessis** [comment]
Hi Abderrahmane,

It looks like you have a caching issue here (most webservers cache images in a Production setting, and I think that is why you are always getting the same captcha image).

Please can you try this:

Change your route to not use the /img base path:

Router::connect('/captchas/captcha.jpg', array('controller' => 'contacts', 'action' => 'captcha'));

Then in your View, change the Image url to:

echo $this->Html->image('/captchas/' . $captcha . '.jpg', array('id' => $captcha));

This should eliminate the caching on the /img assets.

Please can you let me know how this goes?

Regards,
Donovan

**Comment by 218labs** [comment]
Hi,
it works now after adding the prefix language to the action.

Regards,
Abderrahmane
